lib/internal/modules/esm/fetch_module.js

```diff
@@ -248,10 +245,10 @@ allowList.addRange('127.0.0.1', '127.255.255.255');
 async function isLocalAddress(hostname) {
   try {
     if (
-      StringPrototypeStartsWith(hostname, '[') &&
-      StringPrototypeEndsWith(hostname, ']')
+      hostname[0] === '[' &&
+      hostname[hostname.length - 1] === ']'
     ) {
-      hostname = StringPrototypeSlice(hostname, 1, -1);
+      hostname = hostname.slice(1, -1);
```

Comment:
this isn't viable; calling dot methods isn't safe, which is what primordials are for. this line should be reverted.